internal/model/api.go

```diff
@@ -1,4 +1,4 @@
-package controller
+package model
```

Comment:
the idea is good,
but the validation is tricky.. it should belong to controller instead of model

Reply:
Why?

Reply:
validation function can go in the model?

Reply:
if it is MVC, models should avoid any business logic, and it is the responsibility of controller to validate the inputs (this is the concept but not practical sometimes)

Reply:
I changed it to type, I agree.